bin/generate.js

```diff
@@ -0,0 +1,75 @@
+#!/usr/bin/env node
+(function (_, request, xmlbuilder, fs) {
+    'use strict';
+
+    module.exports = function (environment, callback) {
+        var pathFor, entryFor, types, fixedUrls, config;
+
+        // Get the URL path for the given object.
+        pathFor = (object) => '/' + types[object.type] + '/detail/' + object.object_id;
+
+        // Get the <url> element entry for the given parameters (path and priority).
+        entryFor = (params) => ({
+            loc: { '#text': config.frontend.protocol + '://' + config.frontend.hostname + params.path },
+            priority: { '#text': params.priority }
+        });
+
+        // Load configuration.
+        types = require('../config/types.json');
+        fixedUrls = require('../config/fixed-urls.json');
+        config = require('../config/config.' + environment + '.json');
+
+        // Load the data and write it to `sitemap.xml`.
+        request(
+            config.service.protocol + '://' + config.service.username + ':' + config.service.password + '@' + config.service.hostname + '/service.php/simple/sitemap_search?q=*&noCache=1',
+            function (err, response) {
+                var data;
+
+                if (err) {
+                    return callback(err);
+                }
+
+                data = JSON.parse(response.body);
```

Comment:
If the response isn't JSON won't this vomit here? (although it would presumably vomit in a predictable and easy to way)